### PR TITLE
Ui table everything

### PR DESCRIPTION
**Pull request by Calvinp**
Deletes all SimpleTables and ServerSideTables so that UITable is the only remaining table component

**Simple Tables**
- [x] taskDetail/ShellCommands
- [x] taskDetail/TaskAlerts
- [x] taskDetail/TaskHealthchecks
- [x] taskDetail/TaskHistory
- [x] taskDetail/TaskLbUpdates
- [x] taskDetail/TaskS3Logs
- [x] Webhooks
- [x] deployDetail/DeployDetail
- [x] deployDetail/ActiveTasksTable
- [x] MachinesPage

**Server Side Tables**
- [x] deployDetail/DeployDetail
- [x] requestDetail/DeployHistoryTable
- [x] requestDetail/RequestHistoryTable
- [x] requestDetail/TaskHistoryTable
- [x] TaskSearch

### CODE REVIEW

**Comment by Calvinp** [comment]
cc @tpetr @kwm4385 @wolfd

File: SingularityUI/app/components/taskDetail/TaskS3Logs.jsx [hunk 1]
```diff
@@ -1,36 +1,53 @@
...
-            <tr key={index}>
-              <td>
-                <a className="long-link" href={data.getUrl} target="_blank" title={data.key}>
-                    {Utils.trimS3File(data.key.substring(data.key.lastIndexOf('/') + 1), props.taskId)}
-                </a>
-              </td>
-              <td>{Utils.humanizeFileSize(data.size)}</td>
-              <td>{Utils.absoluteTimestamp(data.lastModified)}</td>
-              <td className="actions-column">
-                <a href={data.getUrl} target="_blank" title="Download">
```
Comment (Calvinp) [inline SingularityUI/app/components/taskDetail/TaskS3Logs.jsx]: This was a bug, it should have been `data.downloadUrl`. If this PR is for some reason rejected that will still need to be changed.

File: SingularityUI/app/actions/api/webhooks.es6 [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { buildApiAction, buildJsonApiAction } from './base';
 
 export const FetchWebhooks = buildApiAction(
   'FETCH_WEBHOOKS',
-  {url: '/webhooks'}
```
Comment (Calvinp) [inline SingularityUI/app/actions/api/webhooks.es6]: This was also a bug, and will need to be fixed independently if this PR gets rejected.

File: SingularityUI/app/components/common/table/UITable.jsx [hunk 3]
```diff
@@ -229,12 +235,13 @@ class UITable extends Component {
     const rowsPerPage = this.props.rowChunkSize;
     if (this.props.paginated && numRows > rowsPerPage) {
       const numPages = Math.ceil(numRows / rowsPerPage);
+      const showFirstAndLast = this.props.maxButtons ? numPages > this.props.maxButtons : numPages > 10;
```
Comment (zdhickman) [inline SingularityUI/app/components/common/table/UITable.jsx]: If you want to keep the magic number 10 here, you can shorten to:

`const showFirstAndLast = numPages > (this.props.maxButton || 10);`

or, you could create a default value of 10 for `this.props.maxButton`

**Comment by tpetr** [comment]
@wolfd can you review these tweaks to UITable?

**Comment by Calvinp** [comment]
Note that I haven't implemented default sorting on infinite scroll tables. That's because when I was doing this I encountered a bunch of bugs that have to be fixed in the `decaf` branch. I'd be happy to circle back to that after those are fixed. For now, we're not using that functionality, so I don't think it has to be a blocker for this PR.

**Comment by Calvinp** [comment]
On the other hand, infinite scrolling doesn't currently work (on this branch or `decaf`). The table refuses to show more than some number of components. So I might be back to this sooner rather than later.

Where it looks like the bottom of the page in this screenshot, it is - and you can't scroll past it.

screenshot 2016-07-28 15 35 40

File: SingularityUI/app/components/common/table/UITable.jsx [hunk 4]
```diff
@@ -12,19 +11,35 @@ class UITable extends Component {
...
 
     this.handlePageChange = this.handlePageChange.bind(this);
   }
 
   componentWillReceiveProps(nextProps) {
     this.updateSort(nextProps.data, this.state.sortBy, this.state.sortDirection);
+    if (nextProps.isFetching) {
+      return;
+    }
+    if (this.isServerSideTable() && _.isEmpty(nextProps.data) && this.state.chunkNum > 1) {
```
Comment (wolfd) [inline SingularityUI/app/components/common/table/UITable.jsx]: I think I `isApiPaginated()` might be a be a better method name.

File: SingularityUI/app/components/common/table/UITable.jsx [hunk 5]
```diff
@@ -12,19 +11,35 @@ class UITable extends Component {
...
+      this.fetchDataFromApi(this.state.chunkNum - 1, this.state.rowChunkSize, this.state.sortBy);
+      this.setState({ pastEnd: true });
+    } else if (this.isServerSideTable() && (this.state.pastEnd || nextProps.data.length < this.state.rowChunkSize)) {
+      this.setState({pastEnd: false, lastPage: true});
```
Comment (wolfd) [inline SingularityUI/app/components/common/table/UITable.jsx]: I know we don't lint spacing on object literals, but I think we should maintain consistency here

File: SingularityUI/app/components/common/table/UITable.jsx [hunk 6]
```diff
@@ -37,12 +52,51 @@ class UITable extends Component {
     rowChunkSize: 30,
     defaultSortBy: undefined,
     defaultSortDirection: UITable.SortDirection.DESC,
-    asyncSort: false
+    asyncSort: false,
+    maxButtons: 10
```
Comment (wolfd) [inline SingularityUI/app/components/common/table/UITable.jsx]: `maxPaginationButtons`?

File: SingularityUI/app/components/common/table/UITable.jsx [hunk 7]
```diff
@@ -249,12 +351,14 @@ class UITable extends Component {
 
   renderWaypoint() {
     return (
-      <tr key={'waypoint'}>
+      <tr key="waypoint">
```
Comment (wolfd) [inline SingularityUI/app/components/common/table/UITable.jsx]: 👍

File: SingularityUI/app/components/common/table/UITable.jsx [hunk 8]
```diff
@@ -343,14 +451,23 @@ UITable.propTypes = {
...
+  rowChunkSizeChoices: PropTypes.arrayOf(PropTypes.number),
+  maxPaginationButtons: PropTypes.number,
   defaultSortBy: PropTypes.string,
   defaultSortDirection: PropTypes.oneOf([
     UITable.SortDirection.ASC,
     UITable.SortDirection.DESC
   ]),
   className: PropTypes.string,
   asyncSort: PropTypes.bool,
+  fetchDataFromApi: PropTypes.func, // (page, numberPerPage, sortBy) -> Promise // Makes this table server-side - no sorting
```
Comment (wolfd) [inline SingularityUI/app/components/common/table/UITable.jsx]: update comment for new name?

**Comment by wolfd** [comment]
LGTM!

**Comment by wolfd** [comment]
About infinite scrolling being broken: I feel like I've fixed that a few times, and it keeps being broken. Weird and annoying. I see that you set the scroll ancestor to the window, did that fix anything?

**Comment by Calvinp** [comment]
That in conjunction with changing the key of the waypoint. The problem was that the waypoint would send an `enter` event, then continue thinking it was in the viewport and never send another `enter` event. By making a key that changes, React makes a completely new waypoint each time, so that problem was alleviated. 

The problem with that was that when the scroll ancestor was the table cell, the new waypoint thought it was always in the viewport (I don't know why that happened). So it would keep sending `enter` events, and the table would keep rendering more and more. By making the ancestor the window, it realized that it was below the screen and didn't fire the `enter` event until it was actually time to fire it.

**Comment by tpetr** [comment]
LGTM

**Pull request merged**